specs/004-enhance-the-casting/contracts/promotion_matrix_api.md

```
@@ -0,0 +1,374 @@
+
+---
+
+### `PromotionMatrix::get_promotion_rule(&self, from: &IrType, to: &IrType) -> Option<&PromotionRule>`
+
+**Purpose**: Retrieves promotion rule for a type pair
+
+**Contract**:
+- **Preconditions**: `from` and `to` must be valid `IrType` variants
+- **Postconditions**: 
+  - Returns `Some(&PromotionRule)` for all 13 × 13 fundamental type pairs
+  - Returns `None` for unsupported types (Pointer, Custom, Struct, Void)
```

markdownlint-cli2 (0.18.1)

Trailing spaces
Expected: 0 or 2; Actual: 1

(MD009, no-trailing-spaces)